### ISSUE TITLE
(S.Complexes).union(Interval(1, 2)) is returned un-evaluated

### ISSUE THREAD

**Issue by gxyd** [sympy/sympy#9969]
```
>>> (S.Complexes).union(Interval(1, 2))
[1, 2] ∪ ℂ    # this should return ℂ
```

This is because of [this](https://github.com/sympy/sympy/blob/master/sympy/sets/fancysets.py#L906) line.

This should be changed from `is S.Reals` -> `.is_subset(S.Reals)`

@aktech can you please verify?

**Comment by aktech** [comment]
>This should be changed from is S.Reals -> .is_subset(S.Reals)

Yes, that would be correct. Thanks for this catch.

**Comment by gxyd** [comment]
fixed by #9977
